pkg/cli/admin/release/extract_tools.go

```diff
@@ -297,16 +297,159 @@ func (o *ExtractOptions) extractCommand(command string) error {
 	infoOptions := NewInfoOptions(o.IOStreams)
 	infoOptions.SecurityOptions = o.SecurityOptions
 	infoOptions.FileDir = o.FileDir
-	release, err := infoOptions.LoadReleaseInfo(o.From, false)
-	if err != nil {
-		return err
+
+	var releaseFromRefErr error
+	targetRelease := targetReleaseInfo{}
+	if len(o.SecurityOptions.ICSPFile) > 0 {
```

Comment:
I thought this entire logic of picking file first and than using server or nothing should be part of the library-go code, why repeating similar logic here? As a consumer of the library-go code I'll pass both icspfile and flag saying to read it from cluster and library-go will know the priority and which to pick.

Reply:
It's entirely transparent for consumer.

Reply:
Yes, thanks, I missed this - this was leftover from previous implementation - I've cleaned this up and pushed the update :bowing_woman:  All the icsp gathering happens in lib-go.